test/contrib/rack/middleware_test.rb

```diff
@@ -99,6 +99,18 @@ def test_request_middleware_get_with_request_uri_and_quantize_option
     assert_nil(span.parent)
   end
 
+  def test_request_middleware_get_with_urls_in_headers
+    # ensure the Rack request is properly traced
+    get '/success?foo=bar', {}, 'REFERER' => 'http://example.com/x=y'
```

Comment:
In order to get headers tagging properly I think we have to make a couple of changes here:

```ruby
# The 'Referer'/'Location' header is not a default header
# Need to configure Rack to add them as tags.
Datadog.configure do |c|
  c.use :rack, headers: { request: ['Referer'] , response: ['Location'] }
end

# The 3rd parameter is the Rack Env.
# Headers get reformatted to `HTTP_X`, so to emulate it:
get '/success?foo=bar', {}, 'HTTP_REFERER' => 'http://example.com/x=y'
```